pkg/controller/seldondeployment/seldondeployment_controller.go

```diff
@@ -699,51 +403,68 @@ func createComponents(r *ReconcileSeldonDeployment, mlDep *machinelearningv1alph
 				deploy.Spec.Replicas = &p.Replicas
 			}
 
-			// Add service orchestrator to first deployment if needed
-			if j == 0 && !hasSeparateEnginePod {
-				engineContainer, err := createEngineContainer(mlDep, &p, engine_http_port, engine_grpc_port)
-				if err != nil {
-					return nil, err
-				}
-				deploy.Labels[machinelearningv1alpha2.Label_svc_orch] = "true"
-
-				deploy.Spec.Template.Spec.Containers = append(deploy.Spec.Template.Spec.Containers, *engineContainer)
-				//deploy.Spec.Template.Spec.ServiceAccountName = getEnv("ENGINE_CONTAINER_SERVICE_ACCOUNT_NAME", "seldon")
-				//deploy.Spec.Template.Spec.DeprecatedServiceAccount = deploy.Spec.Template.Spec.ServiceAccountName
-				//deploy.Spec.Template.Annotations = map[string]string{}
-				if deploy.Spec.Template.Annotations == nil {
-					deploy.Spec.Template.Annotations = make(map[string]string)
-				}
-				//overwrite annotations with predictor annotations
-				for _, ann := range p.Annotations {
-					deploy.Spec.Template.Annotations[ann] = p.Annotations[ann]
+			// create services for each container
+			for k := 0; k < len(cSpec.Spec.Containers); k++ {
+				var con *corev1.Container
+				con = utils.GetContainerForDeployment(deploy, cSpec.Spec.Containers[k].Name)
```

Comment:
It's a pointer thing. We want the reference to the container on the created Deployment which at this point could be a copy of the spec from the SeldonDeployment rather than the exact same go struct.